### PR TITLE
feat(oaf-discriminators): support discriminators for oneof/anyof

### LINKED ISSUE #31: Add Discriminators substitution handling for oneOf/anyOf

### PR DESCRIPTION
**Pull request by asadali214**
This PR adds support for discriminators in oneOf/anyOf cases using discriminator substitutions parameter in JsonHelper.
It also contains new unit tests to verify this feature.

Closes #31

NOTE: we can not merge this PR before releasing https://github.com/apimatic/jsonmapper/pull/38, and https://github.com/apimatic/jsonmapper/pull/40
NOTE: This PR adds breaking change in the constructor of JsonHelper and hence require Major version bump while releasing

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "ext-dom": "*",
     "ext-libxml": "*",
     "apimatic/core-interfaces": "^0.1.0",
-    "apimatic/jsonmapper": "^3.0.4",
+    "apimatic/jsonmapper": "dev-37-oaf-discriminator-support",
```
Comment (asadali214) [inline composer.json:20]: This will be reverted after releasing: https://github.com/apimatic/jsonmapper/pull/38

**Review by sufyankhanrao — APPROVED** [review]
LGTM

**Review by sufyankhanrao — APPROVED** [review]
LGTM

File: composer.json (hunk 1 above)
Comment (asadali214) [reply composer.json]: Updated to ^3.1.1

**Review by sufyankhanrao — APPROVED** [review]
approved

**Pull request merged**